internal/registryutil/auth.go

```diff
@@ -18,16 +18,14 @@ package registryutil
 import (
 	"context"
 
-	"oras.land/oras-go/v2"
 	"oras.land/oras-go/v2/registry/remote"
 	"oras.land/oras-go/v2/registry/remote/auth"
 )
 
 // WithScopeHint adds a hinted scope to the context.
-func WithScopeHint(ctx context.Context, target oras.Target, actions ...string) context.Context {
+func WithScopeHint(ctx context.Context, target any, actions ...string) context.Context {
```

Comment:
Just curious, why `any`?

Reply:
Remove the type-restriction makes function more easy to use because there is no need to do the check during compile-time. 

One example: I can call `remote.NewRepository` and assign the returned value to a `content.Deleter` and pass it as `target` here. If `target` is restricted to `oras.Target`, then caller need to do extra type conversion.